app/src/main/java/com/fmsys/snapdrop/JavaScriptInterface.java

```diff
@@ -230,18 +230,67 @@ public static String getSendTextDialogWithPreInsertedString(final String text) {
     }
     public static String initialiseWebsite() {
         return "javascript: " +
+                //change ReceiveTextDialog._onCopy to connect to JavaScriptInterface
+                "ReceiveTextDialog.prototype._oC = ReceiveTextDialog.prototype._onCopy;" +
+                "ReceiveTextDialog.prototype._onCopy = function(){" +
+                "               this._oC();" +
+                "               SnapdropAndroid.copyToClipboard(this.$text.textContent);" +
+                "            };" +
+                //change PeerUI.setProgress(progress) to connect to JavaScriptInterface
+                "PeerUI.prototype.sP = PeerUI.prototype.setProgress;" +
+                "PeerUI.prototype.setProgress = function(progress){" +
+                "               SnapdropAndroid.setProgress(progress);" +
```

Comment:
Crashes currently, as it's not yet implemented

Reply:
yea true! i merged my patch-2 to fix